compiler/rustc_lint/src/types.rs

```diff
@@ -157,13 +164,19 @@ fn lint_overflowing_range_endpoint<'tcx>(
     };
     let Ok(start) = cx.sess().source_map().span_to_snippet(eps[0].span) else { return false };
 
-    use rustc_ast::{LitIntType, LitKind};
-    let suffix = match lit.node {
-        LitKind::Int(_, LitIntType::Signed(s)) => s.name_str(),
-        LitKind::Int(_, LitIntType::Unsigned(s)) => s.name_str(),
-        LitKind::Int(_, LitIntType::Unsuffixed) => "",
-        _ => bug!(),
+    let suffix = if let Some(cast_ty) = cast_ty {
+        let Ok(ty) = cx.sess().source_map().span_to_snippet(cast_ty.span) else { return false };
+        format!(" as {}", ty)
```

Comment:
Why do we need to change `as` at all? can't we instead change `suggestion: struct_expr.span,` to something like `struct_expr.span.shrink_to_lo().to(eps[1].expr.span)` (i.e. suggest to change the `0..256` part, but not the `as ...`)?

Reply:
Got it! Thanks!

Reply:
I tried, but the suggestions don't work well if parentheses appear (suggestions will break parentheses).
In the latest commit, I separated the suggestions for the `=` and the literal.